Applications/spire/source/common/local_technicals_model.cpp

```diff
@@ -0,0 +1,104 @@
+#include "spire/spire/local_technicals_model.hpp"
+
+using namespace boost;
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+LocalTechnicalsModel::LocalTechnicalsModel(Security security)
+    : m_security(std::move(security)),
+      m_volume(0) {}
+
+void LocalTechnicalsModel::set_close(Money price) {
+  m_close = price;
+  m_close_signal(*m_close);
+}
+
+void LocalTechnicalsModel::update(const TimeAndSale& time_and_sale) {
+  if(!m_last_price.is_initialized() || time_and_sale.m_price != m_last_price) {
+    m_last_price = time_and_sale.m_price;
+  }
+  if(!m_high.is_initialized() || time_and_sale.m_price > *m_high) {
+    m_high = time_and_sale.m_price;
+  }
+  if(!m_low.is_initialized() || time_and_sale.m_price < *m_low) {
+    m_low = time_and_sale.m_price;
+  }
+  m_volume += time_and_sale.m_size;
+  if(!m_open.is_initialized()) {
+    m_open = time_and_sale.m_price;
+    m_open_signal(*m_open);
+  }
+  if(*m_last_price != time_and_sale.m_price || *m_low == *m_high) {
```

Comment:
This does not look correct, what if ```m_last_price == time_and_sale.price && m_low == m_high``` then you will be emitting signals that are not necessary.

Add appropriate unit tests to test and correct this issue.